src/checkout/components/Steps.tsx

```diff
@@ -68,14 +71,14 @@ const Steps: React.FC<{
 }> = ({ checkout, step: currentStep, token, children }) => {
   const { lines: cardLines } = React.useContext(CartContext);
   const { data: variantsProducts } = useVariantsProducts({
-    ids: cardLines ? cardLines.map(line => line.variantId) : [],
+    ids: cardLines?.map(line => line.variantId) || [],
```

Comment:
for `x.map(...)`, what happens if `x` is `undefined` or `null`, the return will just be undefined?

Reply:
yes . if x is null or undefined in that case it will return undefined as return value.
Example
const data = x?.map((item) => ---) (here if x is null or undefined)
console.log(data) then return data equal to undefined